op-service/sources/receipts_rpc.go

```diff
@@ -112,6 +112,144 @@ func (f *RPCReceiptsFetcher) FetchReceipts(ctx context.Context, blockInfo eth.Bl
 	return
 }
 
+// BatchFetchReceipts fetches receipts for multiple blocks and transaction hashes.
+// it selects the best RPC method, and then uses BatchCallContext to fetch all receipts at once.
+// The results are individually validated against the blockInfos provided
+func (f *RPCReceiptsFetcher) BatchFetchReceipts(ctx context.Context, blockInfos []eth.BlockInfo, txHashes [][]common.Hash) ([]types.Receipts, error) {
```

Comment:
Using the more optimal RPC methods is nice, but I am less convinced of reimplementing the calls to these, just to get the requests into a JSON-RPC batch. I thought just using the existing receipt fetching for multiple blocks in parallel could be most of the performance improvement.

If we have a JSON-RPC connection over a websocket, then the connection already stays open, and internally in op-geth there is nothing that I think optimizes handling of a batch of requests vs. handling of individual requests in parallel.

Reply:
> Using the more optimal RPC methods is nice, but I am less convinced of reimplementing the calls to these, just to get the requests into a JSON-RPC batch. I thought just using the existing receipt fetching for multiple blocks in parallel could be most of the performance improvement.

Do you mean this with regards to the *entire* BatchFetch method here, or just the extended (alchemy, erigon, etc) calls? If the second, that sounds great, I would love to reduce complexity.

> If we have a JSON-RPC connection over a websocket
good question... do we?